lib/reports/BackupReportManager.js

```diff
@@ -157,78 +203,75 @@ class BackupReportManager {
-      instanceId: instanceId,
-      createdAt: {
-        '$gte': startTime,
-        '$lt': endTime
+  //Count number of entries present in jobrundetails collection which is how many times scheduler was run/backups were triggered
+  static getBackupTriggerCount(instanceId, startTime, endTime) {
+    const criteria = {
+      type: CONST.JOB.SCHEDULED_BACKUP,
+      'data.instance_id': instanceId,
+      'response.delete_backup_status.instance_deleted': {
+        $ne: true
       },
-      'response.guid': {
-        '$exists': true,
-        '$ne': null
-      }
-    };
-    const group = {
-      _id: '$response.guid',
       createdAt: {
-        $first: '$createdAt'
+        $gt: startTime,
+        $lt: endTime
       }
     };
-    const sort = {
-      createdAt: CONST.REPORT_BACKUP.SORT.ASC
-    };
-    const aggregateCriteria = [{
-      $match: match
-    }, {
-      $group: group
-    }, {
-      $sort: sort
-    }];
-    return Repository.aggregate(CONST.DB_MODEL.EVENT_DETAIL, aggregateCriteria);
+    return Repository.count(CONST.DB_MODEL.JOB_RUN_DETAIL, criteria);
   }
 
-  //Aggregated documents of all the succeeded and failed backup entries, sorted first by metric then by create time
+  // Documents of all the succeeded and failed backup entries, sorted first by metric then by create time
   static getBackupResult(instanceId, startTime, endTime) {
-    const match = {
-      eventName: 'create_backup',
-      instanceId: instanceId,
-      metric: {
-        '$ne': config.monitoring.inprogress_metric
-      },
-      createdAt: {
-        '$gte': startTime,
-        '$lt': endTime
-      },
-      'request.backup_guid': {
-        '$exists': true,
-        '$ne': null
+    function getBackupResult(resultList, offset, modelName, searchCriteria, paginateOpts) {
+      if (offset < 0) {
+        return Promise.resolve();
       }
-    };
-    const group = {
-      _id: '$request.backup_guid',
-      metric: {
-        $first: '$metric'
+      if (!paginateOpts) {
+        paginateOpts = {
+          records: config.mongodb.record_max_fetch_count,
+          offset: offset
+        };
+      } else {
+        _.chain(paginateOpts)
+          .set('offset', offset)
+          .value();
+      }
+      return Repository.search(modelName, searchCriteria, paginateOpts)
+        .then((result) => {
+          resultList.push.apply(resultList, result.list);
+          return getBackupResult(resultList, result.nextOffset, modelName, searchCriteria, paginateOpts);
+        });
+    }
+
+    const criteria = {
+      searchBy: {
+        eventName: 'create_backup',
+        instanceId: instanceId,
+        metric: {
+          '$ne': config.monitoring.inprogress_metric
+        },
+        createdAt: {
+          '$gte': startTime,
+          '$lt': endTime
+        }
       },
-      createdAt: {
-        $first: '$createdAt'
+      projection: {
+        metric: 1,
+        'request.backup_guid': 1,
```

Comment:
by having a projection on backup guid, will it filter out results which dont have backup guid in request? For ex, run the below query:
```
db.eventdetails.find({eventName : 'create_backup', metric: 1, 'response.status' : {'$eq': 500}}).pretty();
```
you will see many backups which are failed due to timeouts i guess which dont have backup guids. I guess these are the entries which might come in as trigerred count ? so may be if you tweak projection here you might get the counts properly from eventDetails itself? Can you tweak projection to the below and try?
```
 ...
  request: 1
  response: 1
...
}
```

Reply:
In case backup_guid is not present in some document then resulted list will have empty value for backup_guid key but this document will be part of result